include/Basic/SourceLocation.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef GLU_SOURCE_LOCATION_HPP
+#define GLU_SOURCE_LOCATION_HPP
+
+#include <cstdint>
+
+namespace glu {
+
+class SourceManager;
+
+///
+/// @class FileID
+/// @brief Contains an opaque identifier representing a file, used by the
+/// SourceManager.
+///
+class FileID {
+    friend class SourceManager;
+
+private:
+    /// The opaque identifier for the file.
+    int _id = 0;
+
+private:
```

Comment:
why write private if everything is private? it's a class so everything is private by default